lib/ckb/serializers/dyn_vec_serializer.rb

```diff
@@ -32,7 +32,10 @@ def offsets
         offsets = []
         items.each.with_index do |_item, index|
           offsets << offset0 and next if index == 0
-          offsets << offset0 += item_serializer.new(items[index - 1]).capacity
+
+          sum_of_prev_offsets = offsets[0..index - 1].reduce(0, :+)
```

Comment:
It's a waste of full round calculation. Instead, keep a cursor (`previous_offset`), then `new_offset = previous_offset + current_serializer.capacity; previous_offset = new_offset;`.

Reply:
thanks, fixed.